src/fontations/lib.rs

```diff
@@ -612,20 +617,35 @@ impl ColorPainter for HbColorPainter<'_> {
             self.fill(brush);
         }
         self.pop_transform();
-        self.pop_clip();
+        unsafe {
+            hb_paint_pop_clip(self.paint_funcs, self.paint_data);
+        }
         self.pop_transform();
     }
     fn push_clip_glyph(&mut self, glyph_id: GlyphId) {
+        if self.clip_depth < 64 {
+            self.is_glyph_clip |= 1 << self.clip_depth;
+            self.clip_depth += 1;
+        } else {
+            return;
+        }
         unsafe {
+            hb_paint_push_inverse_font_transform(self.paint_funcs, self.paint_data, self.font);
             hb_paint_push_clip_glyph(
                 self.paint_funcs,
                 self.paint_data,
                 glyph_id.to_u32() as hb_codepoint_t,
                 self.font,
             );
+            hb_paint_push_font_transform(self.paint_funcs, self.paint_data, self.font);
         }
     }
     fn push_clip_box(&mut self, bbox: BoundingBox) {
+        if self.clip_depth < 64 {
+            self.clip_depth += 1;
```

Comment:
I think you do need to clear the bit here since it might have been set at this depth by an earlier subgraph.

Reply:
Oh right.